File: FragmentApplication/app/src/main/java/com/example/fragmentapplication/MainActivity.java
package com.example.fragmentapplication;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.FragmentManager;

import android.os.Bundle;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity implements MainFragment.FragmentInteractionListener {

    FragmentManager fragmentManager;
    SecondFragment secondFragment;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        secondFragment = new SecondFragment();
        fragmentManager = getSupportFragmentManager();
        fragmentManager
                .beginTransaction()
                .add(R.id.first_container, new MainFragment(), "first")
                .addToBackStack("first")
                .commit();

//        fragmentManager
//                .beginTransaction()
//                .replace(R.id.container, new SecondFragment())
//                .commit();
    }


    @Override
    public void onInteract(int type) {
//        SecondFragment secondFragment = (SecondFragment) fragmentManager.findFragmentById(R.id.first_container);
        if (type == 1)
            fragmentManager
                    .beginTransaction()
                    .replace(R.id.first_container, new SecondFragment())
                    .addToBackStack("second")
                    .commit();
        else if (type == 2)
            fragmentManager
                    .beginTransaction()
                    .replace(R.id.first_container, new ThirdFragment())
                    .addToBackStack("third")
                    .commit();
        else {
//            fragmentManager.popBackStack();
            fragmentManager.popBackStack("first", 0);
            fragmentManager.popBackStack("second", 0);
        }
    }
}

File: FragmentApplication2/app/src/main/java/com/example/fragmentapplication/MainActivity.kt
package com.example.fragmentapplication

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import androidx.navigation.findNavController

class MainActivity : AppCompatActivity() {

    var status = 0
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

//        supportFragmentManager.beginTransaction()
//            .replace(R.id.second_fragment_container, SecondFragment())
//            .addToBackStack("initial")
//            .commit()
    }


    override fun onSupportNavigateUp(): Boolean =
        findNavController(R.id.nav_host_fragment).navigateUp()

//    fun changeFragment(type: Int) {
//        when (type) {
//            1 -> {
//                supportFragmentManager.beginTransaction()
//                    .replace(R.id.second_fragment_container, ThirdFragment())
//                    .addToBackStack("third").commit()
//            }
//            2 -> {
//                supportFragmentManager.beginTransaction()
//                    .replace(R.id.second_fragment_container, FirstFragment())
//                    .addToBackStack("first").commit()
//            }
//        }
//    }

    fun onClick(view: View) {
//        if (status == 0) {
//            supportFragmentManager.beginTransaction()
//                .replace(R.id.second_fragment_container, ThirdFragment())
//                .addToBackStack("third").commit()
//            status = 1
//        } else {
//            supportFragmentManager.popBackStack("third", 0)
//            status = 0
//        }
//
//        val intent = Intent(this, SecondActivity::class.java)
//        intent.putExtra("message", "Hello from MainActivity!")
//        startActivity(intent)
    }
}
